tardis/plasma/properties/nlte_rate_equation_solver.py

```diff
@@ -0,0 +1,290 @@
+import stat
+import pandas as pd
+import numpy as np
+
+from tardis.plasma.properties.base import ProcessingPlasmaProperty
+
+__all__ = [
+    "NLTERateEquationSolver",
+]
+
+
+class NLTERateEquationSolver(ProcessingPlasmaProperty):
+    outputs = ("ion_number_density_nlte", "electron_densities_nlte")
+
+    def calculate(
+        self,
+        gamma,
+        alpha_sp,
+        alpha_stim,
+        coll_ion_coeff,
+        coll_recomb_coeff,
+        partition_function,
+        levels,
+        level_boltzmann_factor,
+        phi,
+        rate_matrix_index,
+        number_density,
+    ):
+
+        (
+            photo_ion_rate,
+            rad_recomb_rate_coeff,
+            coll_ion_coefficient,
+            coll_recomb_coefficient,
+        ) = NLTERateEquationSolver.prepare_ion_recomb_rates_nlte_ion(
```

Comment:
Why not `self` instead of `NLTERateEquationSolver`? `Calculate` is not a static method.

Reply:
prepare_ion_recomb_rates_nlte_ion is a static method. Should I switch it to non static?

Reply:
oh, never mind, I got what you're saying.

Reply:
Done